helpers/remove_empty_mobilome_files.py

```diff
@@ -21,15 +21,20 @@
 
 
 def main(results_folder):
+    count_total = 0
+    count_removed = 0
     pattern = os.path.join(results_folder, "species_catalogue", '**', '*_mobilome.gff')
     all_matching_files = glob.glob(pattern, recursive=True)
     filtered_files = [file for file in all_matching_files if os.path.basename(file).count('_') == 1]
     for mobilome_file in filtered_files:
+        count_total += 1
         file_has_contents = evaluate_file(mobilome_file)
         if not file_has_contents:
+            count_removed += 1
             print("Cleaning {}".format(mobilome_file))
             with open(mobilome_file, "w") as f_out:
                 f_out.write("##gff-version 3\n")
+    print("Removed {} empty mobilome files out of {} total mobilome files".format(count_removed, count_total))
```

Comment:
I understand what this is doing I think.. but just for future versions: 
- we should check why the MAP pipeline is spitting out empty files :eek
- the search by one underscore seems a little fragile.. do you have more exact filenames?
- why rewrite with header only rather than remove?
Sorry I'm potentially missing some logic from other scripts here

Reply:
Thanks a lot! You are right about the filtering process, that's not great. I replaced it with regex, that should be more stable. Thanks for pointing that out!

Yes, ideally MAP should not be outputting these files (they contain the entire genome sequence but no annotation since there isn't one to output). I did flag this but I think it's in the backlog. We do keep the header because we decided for consistency that if there is no annotation (for this tool or others) we will just produce a file like this. This way the number of files for each species rep is always the same regardless of whether or not there was an annotation.